src/video/wayland/SDL_waylandwindow.c

```diff
@@ -44,6 +44,139 @@
 #include <libdecor.h>
 #endif
 
+/* _GNU_SOURCE needed on Linux for memfd_create() and mkostemp() */
+#if defined(__linux__) && !defined(_GNU_SOURCE)
+#define _GNU_SOURCE
+#endif
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <errno.h>
+
+#define COLOR_BUFFER_SIZE 4
+
+static void
+release_shm_buffer(void *data, struct wl_buffer *buffer)
+{
+    struct wl_buffer **buf_ptr = data;
+
+    wl_buffer_destroy(buffer);
+
+    if (buf_ptr) {
+        *buf_ptr = NULL;
+    }
+}
+
+struct wl_buffer_listener buffer_listener = {
+    release_shm_buffer
+};
+
+/* Temp file creation method taken from libdecor and should cover Linux and the various BSDs. */
+struct wl_buffer*
+CreateSolidColorBuffer(SDL_VideoData *vid, const SDL_Color *color)
+{
+    static const char template[] = "/SDL_wayland_shm-XXXXXX";
+
+    const char         *path;
+    char               *name;
+    size_t              name_len;
+    Uint8              *buf_data;
+    struct wl_shm_pool *pool;
+    struct wl_buffer   *buffer = NULL;
+    int                 err;
```

Comment:
If we don't have posix_fallocate():
```
/tmp/SDL/src/video/wayland/SDL_waylandwindow.c: In function ‘CreateSolidColorBuffer’:
/tmp/SDL/src/video/wayland/SDL_waylandwindow.c:86:25: warning: unused variable ‘err’ [-Wunused-variable]
   86 |     int                 err;
```

Reply:
Just have to mark it as SDL_UNUSED to silence the warning.  Will be fixed in the next push.